### ISSUE TITLE
A misleading error is reported for a local or parameter conflicting with a range variable from an outer scope

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#12052]
```
        var res = from a in new[] { 1 }
                  select (Func<int, int>)(a => 1);
```

Observed:
                  select (Func<int, int>)(**a** => 1);
error CS0412: 'a': a parameter or local variable cannot have the same name as a method type parameter

Expected:
An error indicating a conflict with a range variable.

This is going to affect pattern variable declarations and out variable declarations.

**Comment by gafter** [comment]
Range variables are syntactic entities that do not have scope. They are given scope when the semantics of the language have them translated into lambda parameters. So the diagnostic produced is precisely correct.

Having said that, I agree that it would be better worded as you describe, even if that is less precise.

**Comment by gafter** [comment]
Oh, I see it says that there is a conflict with a 'type parameter'. That isn't correct.
